README.md

```diff
@@ -1,34 +1,51 @@
+![Logical Components](/doc/logical-components.png)
 
-[API routes](https://nextjs.org/docs/api-routes/introduction) can be accessed on [http://localhost:3000/api/hello](http://localhost:3000/api/hello). This endpoint can be edited in `pages/api/hello.js`.
+## Architecture
 
-The `pages/api` directory is mapped to `/api/*`. Files in this directory are treated as [API routes](https://nextjs.org/docs/api-routes/introduction) instead of React pages.
+The following endpoints are used so that the SPA uses simple one-liners to perform its OAuth work:
 
-## Learn More
+| Endpoint          | Description                                                                          |
+|-------------------|--------------------------------------------------------------------------------------|
+| POST /login/start | Start a login by providing the request URL to the SPA and setting temporary cookies. |
+| POST /login/end   | Complete a login and issuing secure cookies for the SPA containing encrypted tokens. |
+| GET /userInfo     | Return information from the User Info endpoint for the SPA to display.               |
+| GET /claims       | Return ID token claims such as `auth_time` and `acr`.                                |
+| POST /refresh     | Refresh an access token and rewrite cookies.                                         |
+| POST /logout      | Clear cookies and return an end session request URL.                                 |
 
-To learn more about Next.js, take a look at the following resources:
+For further details see the [Architecture](/doc/Architecture.md) article.
 
-- [Next.js Documentation](https://nextjs.org/docs) - learn about Next.js features and API.
-- [Learn Next.js](https://nextjs.org/learn) - an interactive Next.js tutorial.
+## OAuth Agent Development
 
-You can check out [the Next.js GitHub repository](https://github.com/vercel/next.js/) - your feedback and contributions are welcome!
+See the [Setup](/doc/Setup.md) article for details on productive OAuth Agent development.\
+This enables a test driven approach to developing the OAuth Agent, without the need for a browser.
 
-## Deploy on Vercel
+## End-to-End SPA Flow
 
-The easiest way to deploy your Next.js app is to use the [Vercel Platform](https://vercel.com/import?utm_medium=default-template&filter=next.js&utm_source=create-next-app&utm_campaign=create-next-app-readme) from the creators of Next.js.
+See the below article for details on how to run the end-to-end solution in a browser:
 
-Check out our [Next.js deployment documentation](https://nextjs.org/docs/deployment) for more details.
+- [SPA Code Example](https://curity.io/resources/learn/token-handler-spa-example/)
+
+The end-to-end solution, by default, uses our Express implementation of the OAuth Agent. If you want to run it with this Next.js implementation,
+then a few things would have to be changed manually in the build and deployment scripts.
```

Comment:
add link to build script: `https://github.com/curityio/spa-deployments/blob/main/build.sh#L78`
Provide a `Dockerfile` producing the image `oauthagent-standard:1.0.0` to enable easy integration with the example
Then I believe, there is no need to update any deployment scripts...

Reply:
You would at least have to change the configuration of the reverse proxy as it has to point to this instance of the OAuth Agent. I will leave it like that for now, but we can think of making a generic script that would be able to spin up an environment with any combination of OAuth Agent + OAuth Proxy.

Reply:
In the SPA Code Example, everything is setup locally using docker containers, including the reverse proxy.
If there is a Dockerfile that can be used to create an image to run the nextjs application for the oauth agent within a container, users can simply change the link of the repository of the oauth-agent in the build file and the deploy script will build the image and run the container with the oauth agent.